webapp/src/App.tsx

```diff
@@ -159,6 +154,10 @@ function Layout() {
         <Route path={`/shelter/:id`}>
           <ListWithAdoptDogs />
         </Route>
+        <Route path={`/adoptdogs/:sId/dogs/:id`}>
```

Comment:
maybe try to use full names like shelterId and dogId if there is ambiguity instead of sId and id, especially that not even 'shelter' is present in this route